hawkmoth/parser.py

```diff
@@ -88,25 +88,21 @@ def _comment_extract(tu):
         # instead of accessing the `cursor` property multiple times.
         token_cursor = token.cursor
 
-        # cursors that are 1) never documented themselves, and 2) allowed
-        # between comment and the actual cursor being documented
+        # Cursors that are 1) never documented themselves, and 2) allowed
+        # between the comment and the actual cursor being documented.
         if token_cursor.kind in [CursorKind.INVALID_FILE,
                                  CursorKind.TYPE_REF,
                                  CursorKind.PREPROCESSING_DIRECTIVE,
                                  CursorKind.MACRO_INSTANTIATION]:
             continue
 
-        if cursor is not None and token_cursor == cursor:
-            continue
-
-        cursor = token_cursor
-
-        # Note: current_comment may be None
-        if current_comment is not None and docstring.Docstring.is_doc(current_comment.spelling):
-            comments[cursor.hash] = current_comment
+        # Otherwise the current comment documents _this_ cursor. I.e.: not a top
+        # level comment.
+        if current_comment and docstring.Docstring.is_doc(current_comment.spelling):
+            comments[token_cursor.hash] = current_comment
         current_comment = None
```

Comment:
It bugs me that I can't figure out why I've added the `cursor` variable in the first place! I've got a branch with *all* the history dating back to when I was just toying with using libclang from C, but there's no explanation. I give up. :)